### PR TITLE
feature: make package environment agnostic

### LINKED ISSUE #62: Ideas to unblock usage in deno?
### ✅ Prerequisites

- [x] Did you perform a cursory search of open issues? Is this feature already requested elsewhere?
- [x] Are you reporting to the correct repository (`@magic-sdk/admin`)?

### ✨ Feature Request

[Deno](https://deno.land/) is gaining a lot of traction these days thanks to its security model, web compatibility, etc. 

I'd like to be able to use Deno as my backend with Magic, but currently it doesn't seem to be possible. The main blockers I saw when I briefly tried it were a bunch of exceptions on import from ethereum libraries for node.

I've been looking into a few other authn providers lately as well, including Auth0, SuperTokens, and WorkOS, none of which supported deno officially in the form of SDKs, but their SDKs are just thin wrappers around REST APIs that I could hit directly from deno using fetch to unblock myself.

The SDK for Magic however, is doing a bunch of crypto with ethereum libraries and thus not fully abstracted away behind the rest API.

I'm wondering if it might be feasible to change this however? Maybe some of the crypto operations could be moved behind the Magic provided rest endpoints that the SDK already calls? This would also unblock integration in backend languages/environemnts that Magic hasn't provided SDKs for, of which there are still plenty other than deno.

### PR DESCRIPTION
**Pull request by f5io**
### 📦 Pull Request

This PR makes the `@magic-sdk/admin` package environment agnostic by
- removing the usage of `eth-sig-util` and `ethereumjs-util` that rely heavily on `node` internals.
- using `ethereum-cryptography` and some inlined functionality.
- removing usage of node's `Buffer` in favour of typed arrays.
- shimming `node-fetch` only where no global `fetch` implementation is available.
- building both `commonjs` and `esm` outputs of the package.

### ✅ Fixed Issues

- In theory: fixes #62, by making this package platform agnostic, it should work in `deno` although I have not tested.

### 🚨 Test instructions

All tests pass and coverage is maintained.

I have also created an example of this new `@magic-sdk/admin` running in a `sveltekit` application: https://github.com/f5io/magic-link-cloudflare-demo
Here it is actually deployed on cloudflare workers 🎉 : https://magic-link-demo.trustshare.workers.dev/

### ⚠️ Don't forget to add a [semver](https://semver.org/) label!

- `minor`: New Feature

### CODE REVIEW

**Comment by smithki** [comment]
Your contribution is very appreciated @f5io! Some unfortunate timing that we are _currently_ in the process of moving this SDK source code to the [Magic JS monorepo](https://github.com/magiclabs/magic-js). We expect to have completed the transition next week. At that time we would love to help you translate this PR to the new repo!

File: src/utils/rest.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
-import fetch, { RequestInit } from 'node-fetch';
+import { RequestInit } from 'node-fetch';
```
Comment (smithki) [inline src/utils/rest.ts:1]: This can be updated to `import type {...` now that an async shim is being used instead

**Comment by f5io** [comment]
> Your contribution is very appreciated @f5io! Some unfortunate timing that we are _currently_ in the process of moving this SDK source code to the [Magic JS monorepo](https://github.com/magiclabs/magic-js). We expect to have completed the transition next week. At that time we would love to help you translate this PR to the new repo!

absolutely happy to transition these changes into the new monorepo, just let me know when you are ready 👍

File: src/utils/rest.ts (hunk 1 above)
Comment (f5io) [reply src/utils/rest.ts:1]: i did attempt to change type imports with the following in the tsconfig.json...

```json
...
    "importsNotUsedAsValues": "error",
...
```

however, it seems like your current linting set up does not support this syntax :(

**Comment by smithki** [comment]
Priorities shifted a bit and we'll be delaying the merging of this package into our JS monorepo for now. Just a heads up that I'll be reviewing your change shortly and hopefully we can cut a release by end-of-week!

**Comment by smithki** [comment]
@f5io We have updated our test framework to use Jest in order to align this repository with the code standards in our monorepo. This change does create a few minor conflicts in your PR. Once we can get those sorted, I'm happy to ship this one!

File: src/utils/rest.ts (hunk 1 above)
Comment (smithki) [reply src/utils/rest.ts:1]: We can look at this as a follow-up on our side, in that case!

**Comment by f5io** [comment]
@smithki on it... will sort asap

**Comment by f5io** [comment]
@smithki rebased my branch from your master, everything seems to be passing as intended :+1:

**Comment by liam0215** [comment]
I'd like to bump this as well, I think this'll get me unblocked from using Magic with [Convex](https://www.convex.dev/)

**Comment by 3x071c** [comment]
@smithki Don't want to annoy anybody, but I've been encountering similar issues with [Remix](https://remix.run/) on [Cloudflare Pages](https://pages.cloudflare.com/) (running on dynamic workers functions, which don't run full-blown Node.js for efficiency). Would greatly appreciate having this integrated!

**Comment by smithki** [comment]
:rocket: PR was released in [`v1.4.0`](https://github.com/magiclabs/magic-admin-js/releases/tag/v1.4.0) :rocket:

**Pull request merged**